hotrod-client-decoder/src/main/java/org/infinispan/hotrod/numeric/BranchlessParser.java

```diff
@@ -0,0 +1,133 @@
+      int thisVarintMask = firstOneOnStop ^ (firstOneOnStop - 1);
+      int wholeWithContinuations = wholeOrMore & thisVarintMask;
+      // mix them up as per varint spec while dropping the continuation bits:
+      // 0x7F007F isolate the first byte and the third byte dropping the continuation bits
+      // 0x7F007F00 isolate the second byte and the fourth byte dropping the continuation bits
+      // the second and fourth byte are shifted to the right by 1, filling the gaps left by the first and third byte
+      // it means that the first and second bytes now occupy the first 14 bits (7 bits each)
+      // and the third and fourth bytes occupy the next 14 bits (7 bits each), with a gap between the 2s of 2 bytes
+      // and another gap of 2 bytes after the forth and third.
+      wholeWithContinuations = (wholeWithContinuations & 0x7F007F) | ((wholeWithContinuations & 0x7F007F00) >> 1);
+      // 0x3FFF isolate the first 14 bits i.e. the first and second bytes
+      // 0x3FFF0000 isolate the next 14 bits i.e. the third and forth bytes
+      // the third and forth bytes are shifted to the right by 2, filling the gaps left by the first and second bytes
+      return (wholeWithContinuations & 0x3FFF) | ((wholeWithContinuations & 0x3FFF0000) >> 2);
+   }
+
+   private static int readRawVarint40(ByteBuf buffer, int wholeOrMore) {
+      byte lastByte;
+      if (buffer.readableBytes() == 4 || (lastByte = buffer.getByte(buffer.readerIndex() + 4)) < 0) {
+         throw new CorruptedFrameException("malformed varint.");
+      }
+      buffer.skipBytes(5);
+      // add it to wholeOrMore
+      return wholeOrMore & 0x7F |
+            (((wholeOrMore >> 8) & 0x7F) << 7) |
+            (((wholeOrMore >> 16) & 0x7F) << 14) |
+            (((wholeOrMore >> 24) & 0x7F) << 21) |
+            (lastByte << 28);
+   }
+
+   public static long readRawVarint64(ByteBuf buffer) {
+      if (buffer.readableBytes() <= 4)
+         return readRawVarint32(buffer);
+
+      long wholeOrMore = buffer.getLongLE(buffer.readerIndex());
+      long firstOneOnStop = ~wholeOrMore & (0x80808080L << 32 | 0x80808080L);
+
+      // The value occupies the 9 bytes. We just unroll it and consume the bytes.
+      if (firstOneOnStop == 0) {
+         return readRawVarlong72(buffer, wholeOrMore);
+      }
+
+      // Consume the bytes containing the long.
+      int bitsToKeep = Long.numberOfTrailingZeros(firstOneOnStop) + 1;
+      buffer.skipBytes(bitsToKeep >> 3);
+
+      // Create a mask and create the continuation bytes for decoding.
+      long thisVarlongMask = firstOneOnStop ^ (firstOneOnStop - 1);
+      long continuation = wholeOrMore & thisVarlongMask;
+
+      // We parse it as 8 bytes, following the same state strategy for the integer parsing.
+      // Mix the continuation with bytes 1 and 3, bytes 2 and 4, bytes 5 and 7, bytes 6 and 8.
+      // It creates a 2 bit gap between each append of the 7 bits.
+      continuation = (continuation & 0x7F007F) |
```

Comment:
same as https://github.com/infinispan/infinispan-benchmarks/pull/23/files#diff-290f61a909ed44dd2c2b742af23ebd587ab06b630b0ff494288bfca05dc4100fR19-R32

We could reduce drammatically the amount of operations here